### PR TITLE
User-facing content and codebase doc fixups from transifex translator feedback

### LINKED ISSUE #24366: Transifex translators feedback
This is a tracker for Transifex translators feedback cross-posted from https://www.transifex.com/bitcoin/bitcoin/:

- [ ] https://www.transifex.com/bitcoin/bitcoin/translate/#fr/qt-translation-023x/410183589
https://github.com/bitcoin/bitcoin/blob/03c8c6937ee03787c36ce374c50201f85a0a6c61/src/qt/intro.cpp#L301
> This seems redundant as free space is available space.

- [ ] https://www.transifex.com/bitcoin/bitcoin/translate/#fr/qt-translation-023x/410184314
https://github.com/bitcoin/bitcoin/blob/03c8c6937ee03787c36ce374c50201f85a0a6c61/src/wallet/spend.cpp#L781
> Change as in amount returned from a transaction?

Maybe a better wording could help?

- [ ] https://www.transifex.com/bitcoin/bitcoin/translate/#fr/qt-translation-023x/410184290
https://github.com/bitcoin/bitcoin/blob/03c8c6937ee03787c36ce374c50201f85a0a6c61/src/txdb.cpp#L39
> Rather: "cannot be completed".

### PR DESCRIPTION
**Pull request by jonatack**
Closes #24366.

### CODE REVIEW

**Comment by jonatack** [comment]
(When to use *cannot* vs *can not*: https://writingexplained.org/cannot-or-can-not-difference. The cases when "can not" would be used are somewhat rare and not encountered here.)

**Comment by prayank23** [comment]
> (When to use _cannot_ vs _can not_: https://writingexplained.org/cannot-or-can-not-difference. The cases when "can not" would be used are somewhat rare and not encountered here.)

Based on this link: https://english.stackexchange.com/questions/4510/why-is-cannot-spelled-as-one-word both look okay to me. 

Overall I have no issues with the changes made in this PR.

ACK

File: doc/build-unix.md [hunk 1]
```diff
@@ -254,7 +254,7 @@ Hardening enables the following features:
     randomly located as well.
 
     On an AMD64 processor where a library was not compiled with -fPIC, this will cause an error
-    such as: "relocation R_X86_64_32 against `......' can not be used when making a shared object;"
+    such as: "relocation R_X86_64_32 against `......' cannot be used when making a shared object;"
```
Comment (hebasto) [inline doc/build-unix.md:257]: This string looks being copy-pasted from other source. Should we alter it?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

File: doc/build-unix.md (hunk 1 above)
Comment (jonatack) [reply doc/build-unix.md:257]: Not sure, I thought about this, but the "such as:" preceding seems to make it only an example.

**Review by hebasto — APPROVED** [review]
ACK 4982b329b7d4f0e0343b0fa23b19d1a538eca4d6

**Comment by jonatack** [comment]
(TBH I'd noticed most of this diff before but thought they were too pedantic to propose changing. To my mind this is more to avoid spending further time on it.)

**Comment by hebasto** [comment]
> (TBH I'd noticed most of this diff before but thought they were too pedantic to propose changing. To my mind this is more to avoid spending further time on it.)

Translators forms a good focus-group for Bitcoin Core UI testing :)

**Review by w0xlt — APPROVED** [review]
ACK 4982b32

File: src/wallet/spend.cpp [hunk 2]
```diff
@@ -778,7 +778,7 @@ static bool CreateTransactionInternal(
     }
     else if ((unsigned int)nChangePosInOut > txNew.vout.size())
     {
-        error = _("Change index out of range");
```
Comment (luke-jr) [inline src/wallet/spend.cpp:781]: ACK

**Review by luke-jr — COMMENTED** [review]
I think the current strings are better, and don't see rationale for changing them (exception ACK'd)

**Comment by jonatack** [comment]
> I think the current strings are better, and don't see rationale for changing them (exception ACK'd)

That leaves
- "free space available", which I agree may not bother native speakers (it doesn't bother me, at least) but I can see why it may appear redundant to non-natives / English-as-a-second language speakers (as seen in the referenced issue in the PR description), so we may as well avoid the issue coming up
- "can not", for which there are explanations and links above

**Comment by jonatack** [comment]
Summary: this has 3 ACKs from @prayank23, @hebasto and @w0xlt

File: doc/build-unix.md (hunk 1 above)
Comment (laanwj) [reply doc/build-unix.md:257]: Yes, it looks to me it is a compiler message. Better for search ability to leave it as is.

**Comment by laanwj** [comment]
Code review ACK https://github.com/bitcoin/bitcoin/pull/24367/commits/4982b329b7d4f0e0343b0fa23b19d1a538eca4d6
But would prefer the change in `build-unix.md` to be reverted, it's better to quote error messages verbatim even if they contain typos.

File: doc/build-unix.md (hunk 1 above)
Comment (jonatack) [reply doc/build-unix.md]: Thanks, done.

**Comment by jonatack** [comment]
Updated per review feedback to leave the error example as-is in `doc/build-unix.md`. Thanks!

**Review by hebasto — APPROVED** [review]
re-ACK 48742693acc9de837735674057c9aae2fe90bd1d, only suggested change since my previous [review](https://github.com/bitcoin/bitcoin/pull/24367#pullrequestreview-885938219).

File: src/wallet/spend.cpp [hunk 3]
```diff
@@ -778,7 +778,7 @@ static bool CreateTransactionInternal(
...
+        error = _("Transaction change output index out of range");
```
Comment (MarcoFalke) [inline src/wallet/spend.cpp:781]: Is there any value in translating this? I mean it is not possible to set the change position via the GUI, no?

**Pull request merged**